plugins/signals/src/plugin.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  createApiFactory,
+  createPlugin,
+  discoveryApiRef,
+  identityApiRef,
+} from '@backstage/core-plugin-api';
+import { signalApiRef } from '@backstage/plugin-signals-react';
+import { SignalClient } from './api/SignalClient';
+
+let clientInstance: SignalClient | undefined = undefined;
+
+/** @public */
+export const signalsPlugin = createPlugin({
+  id: 'signals',
+  apis: [
+    createApiFactory({
+      api: signalApiRef,
+      deps: {
+        identity: identityApiRef,
+        discoveryApi: discoveryApiRef,
+      },
+      factory: ({ identity, discoveryApi }) => {
+        if (!clientInstance) {
+          clientInstance = SignalClient.create({
+            identity,
+            discoveryApi,
+          });
+        }
```

Comment:
There's not need for this caching, it's already done by the framework.

Reply:
Oh, didn't know that. Great, will remove this